File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.py
from collections import deque

def solution(cards1, cards2, goal):
    ca1 = deque([c for c in cards1])
    ca2 = deque([c for c in cards2])
    for word in goal:
        if word in ca1:
            if word != ca1[0]:
                return ("No")
            ca1.popleft()
        elif word in ca2:
            if word != ca2[0]:
                return ("No")
            ca2.popleft()
        else:
            return ("No")
    answer = 'Yes'
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

string solution(vector<string> cards1, vector<string> cards2, vector<string> goal) {
    queue <string> q;
    queue <string> q2;
    
    for (const auto &card : cards1) q.push(card);
    for (const auto &card : cards2) q2.push(card);
    
    string answer = "Yes";
    
    for (const auto &g : goal) {
        if (g == q.front()) q.pop();
        else if (g == q2.front()) q2.pop();
        else return "No";
    }
    return answer;
}